rotkehlchen/externalapis/alchemy.py

```diff
@@ -0,0 +1,422 @@
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.errors.price import NoPriceForGivenTimestamp, PriceQueryUnsupportedAsset
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.externalapis.interface import ExternalServiceWithApiKeyOptionalDB
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.history.deserialization import deserialize_price
+from rotkehlchen.history.price import PriceHistorian
+from rotkehlchen.history.types import HistoricalPrice, HistoricalPriceOracle
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.interfaces import HistoricalPriceOracleInterface
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ApiKey, ChainID, ExternalService, Price, Timestamp
+from rotkehlchen.utils.misc import set_user_agent, ts_now
+from rotkehlchen.utils.mixins.penalizable_oracle import PenalizablePriceOracleMixin
+from rotkehlchen.utils.network import create_session
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class Alchemy(
+    ExternalServiceWithApiKeyOptionalDB,
+    HistoricalPriceOracleInterface,
+    PenalizablePriceOracleMixin,
+):
+    default_api_key = ApiKey('Y_nFGh5UCi30NzRWOAam0gsVpQrjDXZX')
+
+    def __init__(self, database: 'DBHandler | None') -> None:
+        ExternalServiceWithApiKeyOptionalDB.__init__(
+            self,
+            database=database,
+            service_name=ExternalService.ALCHEMY,
+        )
+        HistoricalPriceOracleInterface.__init__(self, oracle_name='alchemy')
+        PenalizablePriceOracleMixin.__init__(self)
+        self.session = create_session()
+        set_user_agent(self.session)
+        self.db: DBHandler | None  # type: ignore  # "solve" the self.db discrepancy
+
+    def query_current_price(
+            self,
+            from_asset: AssetWithOracles,
+            to_asset: AssetWithOracles,
+    ) -> Price:
+        """
+        Docs:
+            https://docs.alchemy.com/reference/get-token-prices-by-symbol
+            https://docs.alchemy.com/reference/get-token-prices-by-address
+
+        May raise:
+        - RemoteError if there is a problem querying Alchemy
+        """
+        try:
+            asset_identifier = self._get_asset_identifier(from_asset)
+        except UnsupportedAsset:
+            log.warning(
+                f'Tried to query current price using Alchemy from {from_asset.identifier} '
+                f'to {to_asset.identifier}. But from_asset is not supported in alchemy',
+            )
+            return ZERO_PRICE
+
+        if 'symbol' in asset_identifier:
+            result = self._query(
+                verb='GET',
+                payload={'symbols': asset_identifier['symbol']},
+                endpoint='by-symbol',
+            )
+        else:
+            result = self._query(
+                verb='POST',
```

Comment:
you never call the same endpoint both with get and post right? If so make the verb a variable in self._query depending on the endpoint argument